--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -435,11 +435,11 @@
 -Wstrict-prototypes -Wmissing-prototypes \
 -Wno-long-long -Wno-format-extra-args")
-
+
-	if(BUILD_SHARED_LIBS AND UNIX AND NOT APPLE)
-		check_linker_flag(C "LINKER:--undefined-version" FLAG_UNDEFINED_VERSION)
-		if (FLAG_UNDEFINED_VERSION)
-			target_link_options(LibXml2 PRIVATE "LINKER:--undefined-version")
-		endif()
-		target_link_options(LibXml2 PRIVATE "LINKER:--version-script=${CMAKE_CURRENT_SOURCE_DIR}/libxml2.syms")
-	endif()
+    if(BUILD_SHARED_LIBS AND UNIX AND NOT APPLE AND NOT ANDROID)
+        check_linker_flag(C "LINKER:--undefined-version" FLAG_UNDEFINED_VERSION)
+        if (FLAG_UNDEFINED_VERSION)
+            target_link_options(LibXml2 PRIVATE "LINKER:--undefined-version")
+        endif()
+        target_link_options(LibXml2 PRIVATE "LINKER:--version-script=${CMAKE_CURRENT_SOURCE_DIR}/libxml2.syms")
+    endif()
 endif()
